src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs

```diff
@@ -79,14 +56,58 @@ private bool TryConnect(int timeout, CancellationToken cancellationToken)
             {
                 int errorCode = Marshal.GetLastWin32Error();
 
-                // Handle the possible race condition of someone else connecting to the server 
-                // between our calls to WaitNamedPipe & CreateFile.
-                if (errorCode == Interop.Errors.ERROR_PIPE_BUSY)
+                if (errorCode != Interop.Errors.ERROR_PIPE_BUSY &&
+                    errorCode != Interop.Errors.ERROR_FILE_NOT_FOUND)
+                {
+                    throw Win32Marshal.GetExceptionForWin32Error(errorCode);
+                }
+
+                if (!Interop.Kernel32.WaitNamedPipe(_normalizedPipePath, timeout))
                 {
-                    return false;
+                    errorCode = Marshal.GetLastWin32Error();
+
+                    // Server is not yet created
+                    if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND)
+                    {
+                        return false;
+                    }
+
+                    // The timeout has expired.
+                    if (errorCode == Interop.Errors.ERROR_SUCCESS)
```

Comment:
This PR is changing it from ERROR_SUCCESS to ERROR_SEM_TIMEOUT: 
https://github.com/dotnet/corefx/pull/25197
Should that be done here, too?